From c3b891eeec691753fccdf8e0a4541dcba244ec6b Mon Sep 17 00:00:00 2001
From: Niko Mauno <niko.mauno@vaisala.com>
Date: Thu, 10 Oct 2024 11:06:07 +0300
Subject: [PATCH] Disable installing systemd service file

With this we avoid the the following BitBake error when using some
other init manager than SystemD:

  ERROR: fluentbit-3.1.9-r0 do_package: QA Issue: fluentbit: Files/directories were installed but not shipped in any package:
    /lib
    /lib/systemd
    /lib/systemd/system
    /lib/systemd/system/fluent-bit.service
  Please set FILES such that these items are packaged. Alternatively if they are unneeded, avoid installing them or delete them within do_install.
  fluentbit: 4 installed and not shipped files. [installed-vs-shipped]
  ERROR: fluentbit-3.1.9-r0 do_package: Fatal QA errors were found, failing task.

Upstream-Status: Inappropriate [hack]

Signed-off-by: Niko Mauno <niko.mauno@vaisala.com>
---
 src/CMakeLists.txt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index e24666484..c102806d0 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -534,7 +534,7 @@ if(FLB_BINARY)
     set(SYSTEMD_UNITDIR  /lib/systemd/system)
   endif()
 
-  if(SYSTEMD_UNITDIR)
+  if(FALSE)
     set(FLB_SYSTEMD_SCRIPT "${PROJECT_SOURCE_DIR}/init/${FLB_OUT_NAME}.service")
     configure_file(
       "${PROJECT_SOURCE_DIR}/init/systemd.in"
-- 
2.39.2

